torchvision/transforms/functional_tensor.py

```diff
@@ -411,7 +413,22 @@ def pad(img: Tensor, padding: List[int], fill: int = 0, padding_mode: str = "con
         need_cast = True
         img = img.to(torch.float32)
 
-    img = torch_pad(img, p, mode=padding_mode, value=float(fill))
+    img = torch_pad(img, p, mode=padding_mode)
+
+    if padding_mode == "constant":
+        # The following if/else can't be simplified due to JIT
+        if isinstance(fill, (tuple, list)):
+            fill_img = torch.tensor(fill).to(dtype=img.dtype, device=img.device).view(1, -1, 1, 1)
```

Comment:
nit: can't we create it directly as 
```python
fill_img = torch.tensor(fill, dtype=img.dtype, device=img.device)
```
?

Reply:
No. :( JIT requires it to be behind an if statement. I believe this is because it invokes a different C++ method (The one that receives a list VS a scalar).

Reply:
Sorry I missed that. That's also needed for the scalar case. I believe some of the tests were failing due to `fill` being float and dtype being integer. Casting solves this.

BTW you are welcome to push to the branch if you want to experiment.

Reply:
@ansley FYI this is the kind of weird code one must write to make things JIT-scriptable. Without the explicit if statement, JIT doesn't know how to handle `fill` when scalar vs when list. I believe this has to do with the fact that the C++ implementation ends up calling a different method.